src/hotspot/share/opto/divnode.cpp

```diff
@@ -1518,7 +1518,8 @@ Node* ModFNode::Ideal(PhaseGVN* phase, bool can_reshape) {
   PhaseIterGVN* igvn = phase->is_IterGVN();
 
   bool result_is_unused = proj_out_or_null(TypeFunc::Parms) == nullptr;
-  if (result_is_unused) {
+  bool is_dead = proj_out_or_null(TypeFunc::Control) == nullptr;
+  if (result_is_unused && !is_dead) {
```

Comment:
Might be easier to read when it's flipped but I leave it up to you to decide which one you prefer :-)
```
  bool not_dead = proj_out_or_null(TypeFunc::Control) != nullptr;
  if (result_is_unused && not_dead) {
```

Reply:
I'd agree if I wrote `!not_dead`. But between writing `! a_positive_property` and `a_negative_property`, I'm much less decided.

Reply:
I ended up flipping as suggested because I've seen fonts/colors making the `!` not clearly a non-letter (github, for instance isn't very good at that imo), and sometimes, people might be surprised.